modules/110-istio/templates/cni/daemonset.yaml

```diff
@@ -0,0 +1,175 @@
+{{- if eq $.Values.istio.sidecarInjectionMode "CNIPlugin" }}
+{{- $version := $.Values.istio.globalVersion }}
+{{- $versionInfo := get $.Values.istio.internal.versionMap $version }}
+{{- $imageSuffix := get $versionInfo "imageSuffix" }}
+
+{{- if (.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+---
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: istio-cni-node
+  namespace: d8-{{ .Chart.Name }}
+  {{- include "helm_lib_module_labels" (list $ (dict "app" "istio-cni-node" "workload-resource-policy.deckhouse.io" "every-node")) | nindent 2 }}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: DaemonSet
+    name: istio-cni-node
+  updatePolicy:
+    updateMode: "Initial"
+  resourcePolicy:
+    containerPolicies:
+      - containerName: "install-cni"
+        minAllowed:
+          cpu: 100m
+          memory: 100Mi
+        maxAllowed:
+          cpu: 200m
+          memory: 200Mi
+{{- end }}
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: istio-cni-node
+  namespace: d8-{{ .Chart.Name }}
+  {{- include "helm_lib_module_labels" (list $ (dict "app" "istio-cni-node")) | nindent 2 }}
+spec:
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: istio-cni-node
+  template:
+    metadata:
+      annotations:
+        ambient.istio.io/redirection: disabled
+        prometheus.io/path: /metrics
```

Comment:
Make sure there is PodMonitor and if not — add pls.
These labels aren't needed.